xwayland/xwayland.c

```diff
@@ -29,7 +29,9 @@ struct wlr_xwayland_cursor {
 static void handle_server_destroy(struct wl_listener *listener, void *data) {
 	struct wlr_xwayland *xwayland =
 		wl_container_of(listener, xwayland, server_destroy);
+
 	wlr_xwayland_destroy(xwayland);
+	xwm_destroy(xwayland->xwm);
```

Comment:
Hm. This isn't run on restart. right?

Reply:
You're right, it's not even being called..

Reply:
Ok, new patch. I wanted to avoid adding a new signal but this seems to work for wayfire at least.. for some reason, sway still uses 100% cpu when restarting Xwayland.

Reply:
And that is because my signal handler is only called under wayfire, not sway.. go figure.

Reply:
Should be fixed now, I guess sway uses lazy and I wasn't emitting the signal in that case.